### ISSUE TITLE
Sync bar is not visible for Record Yesterday's Activity screen (RYA)

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#9464]
When you're completing Dailies in the Record Yesterday's Activity (RYA) screen, there's no way to tell when each tick mark has been synced to the server and so you can take unfair damage from some of your clicks not being recorded if you click too quickly.

~Possible solutions include making the normal sync bar that appears at the top of the website visible when the RYA screen is up, or adding a customised sync bar to the RYA screen.~

The solution is to have the sync bar above the overlay in the stack (as decided in the discussion below).

~Labels updated. I've marked this as needing design. @Tressley~

**Comment by Alys** [comment]
Labels updated. I'm marked this as needing design. @Tressley

**Comment by Tressley** [comment]
@Alys -- I don't think I am completely clear on the issue, but from what I'm gathering, I don't think needs design, but rather fixing how we display the user's state.

Are you referring to completing RYA and then seeing the Tasks page while the server is still syncing your information? The sync bar is present in that state, but easy to miss upon the visual impression of the Tasks being presented. I think the first visual correction would be not be displaying the Tasks page in an actionable state until the sync is completed.

**Comment by Alys** [comment]
No, this is about displaying the sync bar while you are marking Dailies
complete within the RYA screen. You need to know when eaoh action you take
has synced, otherwise you risk having some of your Dailies not completed
when cron runs. A sync bar needs to be visible in RYA for the same reasons
it's visible on the tasks page.

**Comment by paglias** [comment]
The sync bar is actually there but covered by the purple background behind so it's almost invisible

**Comment by Alys** [comment]
Thanks @paglias !  It sounds like this just needs an adjustment to the current design. Even knowing the sync bar is there, I can't see it. :)

**Comment by Tressley** [comment]
@paglias / @Alys  -- Is it possible to store completed Dailies from the RYA modal and sync them once the user has clicked the "Start my day" button? In the context of the modal, the user is checking off multiple Dailies in a short amount of time. This is different from the typical flow on the Tasks' page. That doesn't feel like the proper logic.

The way RYA works now is:
- Navigate to Habitica and see loading screen
- Land on the Tasks' page of Habitica
- Small pause before displaying RYA modal
- While checking off Tasks in the modal, I can actively see them being toggled to completed and the sync bar being visible (barely) behind the overlay.
- Click "Start my day" and see my old Tasks' while the page is being synced.
**This is the part of the flow I think is causing more friction** I don't think we should be displaying an old state of the Tasks' page to the user during this flow. I also don't think the user should see any changes on the Task's page while the RYA modal is open. Those changes should only be visualized in the modal.

My proposed solution is to toggle the completed state of a Task from the RYA modal and then sync all of the changes once the user is ready to start their day by using a "syncing" state from the RYA modal.

![image](https://user-images.githubusercontent.com/2686586/32838502-201e863a-c9df-11e7-969c-0d55333e20cd.png)

**Comment by Alys** [comment]
@Tressley Even if that's done, can the visibility of the sync bar be improved first since that would be a small, quick change and would give immediate benefit? What you describe would take more effort and there might be a delay in starting it since there's so much else being worked on.

I'm not sure if storing the user's actions in the RYA modal and doing all the syncs at once at the end is the best solution. As you're marking the tasks complete, you should be seeing the usual notifications for gold, experience, mana, drops, and drops at least requires immediate syncing since that information comes from the server. I'm pretty sure that if the usual notifications weren't appearing as you completed the Dailies, users would feel that their actions weren't being recorded at all or that they weren't getting proper credit for their tasks.

Also I think TheHollidayInn is or will be working on something to improve syncing site-wide, which could make this issue irrelevant, but again that might not be a short-term fix.

Anything that happens after you click "Start my day" is irrelevant to this issue (I agree that it's not good to display the old start of the Tasks page but that's a separate issue). The problem here is that currently, to be sure that your actions in RYA are recorded, you need to be able to see the sync bar as you are completing the Dailies. Making the sync bar visible when RYA is open would be a good improvement right now.

**Comment by Tressley** [comment]
Then the solution is to have the sync bar above the overlay in the stack. 👍

**Comment by Alys** [comment]
@Tressley Thanks! I've added that comment to the top post and updated the labels.
